### PR TITLE
feat: Add dynamic property path for sourceData in SelectWidget

### LINKED ISSUE #34568: Change the top input type to an open field (Select)
Users are confused how to add static data to a select widget: https://fathom.video/calls/109679577

[all details](https://www.notion.so/appsmith/Users-are-confused-how-to-add-static-data-to-a-select-widget-f1763b0dcb854b42b38438b50765d32a)

### PR DESCRIPTION
**Pull request by rahulbarwal**
## Description
Add dynamic property path for sourceData in SelectWidget.

Fixes #34568
_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags="@tag.Widget"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/9712372976>
> Commit: c5521e46382d448715183862aededc142136db0a
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=9712372976&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Widget`

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
	- Enhanced the Select Widget with a new property to support dynamic property paths.
- **Tests**
	- Added new tests to validate the dynamic property paths functionality in the Select Widget.

### CODE REVIEW

File: app/client/src/widgets/SelectWidget/widget/index.tsx [hunk 1]
```diff
@@ -93,11 +93,15 @@ class SelectWidget extends BaseWidget<SelectWidgetProps, WidgetState> {
       labelPosition: LabelPosition.Top,
       labelAlignment: Alignment.LEFT,
       labelWidth: 5,
-      sourceData: [
-        { name: "Blue", code: "BLUE" },
-        { name: "Green", code: "GREEN" },
-        { name: "Red", code: "RED" },
-      ],
+      sourceData: JSON.stringify(
```
Comment (AmanAgarwal041) [inline app/client/src/widgets/SelectWidget/widget/index.tsx:96]: Any reason why have we changed the `sourcedata` to a string when it was an array of objects ? @rahulbarwal

Comment (rahulbarwal) [reply app/client/src/widgets/SelectWidget/widget/index.tsx:96]: This commit explains it: https://github.com/appsmithorg/appsmith/pull/34574/commits/2ec7d53ba24842b94cfd5c5eb84dfb7b01ecd994

Let me know if you have more questions.

**Review by AmanAgarwal041 — DISMISSED** [review]
Looks good to me

**Review by AmanAgarwal041 — APPROVED** [review]

**Pull request merged**